### PR TITLE
Add start_date/end_date support.

### PR DESCRIPTION
**Pull request by goodov**
Cherry-pick https://github.com/brave/brave-variations/pull/605

Confirmed this is working on staging https://github.com/brave/brave-browser/issues/29810#issuecomment-1513278874

### CODE REVIEW

**Review by tmancey — APPROVED** [review]
LGTM++

**Review by kjozwiak — APPROVED** [review]
Uplift into `production` approved after deliberating with @brave/uplift-approvers. @goodov and QA have verified the above on main (staging) as per https://github.com/brave/brave-browser/issues/29810#issue-1672969374 & https://github.com/brave/brave-browser/issues/29810#issuecomment-1513278874.

**Pull request merged**
